src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -82,244 +86,146 @@ enum
 
-    void Reset() override
+    void JustRespawned() override
     {
-        m_bHasShattered = false;
-
-        DoCastSpellIfCan(m_creature, SPELL_STONED, CAST_TRIGGERED);
-        DoCastSpellIfCan(m_creature, SPELL_CONSTRUCT_HITTING_YA, CAST_TRIGGERED);
+        m_hasShattered = false;
+        
+        DoCastSpellIfCan(nullptr, SPELL_STONED, CAST_TRIGGERED);
+        DoCastSpellIfCan(nullptr, SPELL_CONSTRUCT_HITTING_YA, CAST_TRIGGERED);
     }
 
-    void JustReachedHome() override
+    void EnterEvadeMode() override
     {
-        // reset flags if necessary
-        if (m_creature->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE))
-            m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
-
-        DoCastSpellIfCan(m_creature, SPELL_STONED, CAST_TRIGGERED);
-        DoCastSpellIfCan(m_creature, SPELL_CONSTRUCT_HITTING_YA, CAST_TRIGGERED);
+        m_creature->SetRespawnDelay(30, true);
+        m_creature->ForcedDespawn();
     }
 
-    void DamageTaken(Unit* pDoneBy, uint32& uiDamage, DamageEffectType /*damagetype*/, SpellEntry const* spellInfo) override
+    void DamageTaken(Unit* dealer, uint32& damage, DamageEffectType /*damagetype*/, SpellEntry const* spellInfo) override
     {
+        if (damage < 5000)
+            return;
         // ToDo: This may need more research related to spell proc
-        if (m_creature->HasAura(m_bIsRegularMode ? SPELL_BRITTLE : SPELL_BRITTLE_H) && !m_bHasShattered)
+        if (m_creature->HasAura(m_isRegularMode ? SPELL_BRITTLE : SPELL_BRITTLE_H) && !m_hasShattered)
         {
-            if (uiDamage > 5000)
-            {
-                DoCastSpellIfCan(m_creature, SPELL_SHATTER, CAST_TRIGGERED);
-                DoCastSpellIfCan(m_creature, SPELL_STRENGTH_REMOVE, CAST_TRIGGERED);
+            m_creature->RemoveAllAurasOnDeath();
+            DoCastSpellIfCan(nullptr, SPELL_SHATTER, CAST_TRIGGERED);
+            DoCastSpellIfCan(nullptr, SPELL_STRENGTH_REMOVE, CAST_TRIGGERED);
 
-                m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
-                m_creature->GetMotionMaster()->MoveIdle();
-                m_bHasShattered = true;
-            }
-        }
-    }
-
-    void SpellHit(Unit* pCaster, const SpellEntry* pSpell) override
-    {
-        if (pSpell->Id == SPELL_HEAT)
-        {
-            if (SpellAuraHolder* pHeatAura = m_creature->GetSpellAuraHolder(SPELL_HEAT))
-            {
-                if (pHeatAura && pHeatAura->GetStackAmount() == MAX_HEAT_STACKS)
-                    DoCastSpellIfCan(m_creature, SPELL_MOLTEN);
-            }
+            m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
+            m_creature->SetImmuneToPlayer(true);
+            m_creature->SetImmuneToNPC(true);
+            m_creature->GetMotionMaster()->MoveIdle();
+            m_hasShattered = true;
         }
     }
 
-    void UpdateAI(const uint32 /*uiDiff*/) override
+    void UpdateAI(const uint32 diff) override
```

Comment:
Remove the updateai entirely.  You can disable combat movement, melee and put it in a combat script instead of this code.